### PR TITLE
Fix issues with drag and drop over text

### LINKED ISSUE #8115: Block drop zone has no background color
**Describe the bug**
When I drag an image inside an paragraph, I see a drop-zone text, but can't read the text, because the background color is missing. The background in `editor/components/block-drop-zone` now overrides `packages/components/drop-zone`, maybe introduced in #7640.

**Screenshots**
drop-file

### PR DESCRIPTION
**Pull request by jasmussen**
This PR is a followup to #8752, and fixes #8115.

It is the situation in `master` right now that every single block has instructional overlay text when dragging files from the desktop over it. This also means very short paragraphs, separators, lots of different blocks that can neither fit the upload message, nor can really _accept_ any uploads. That last part is key, because we also have a different upload indicator — the blue horizontal line. This indicator shows _where the block you're about to drop will be inserted_. Because you can drop an image file on top of a paragraph, but it will show up either _before_ the paragraph, or _after_ the paragraph, not _inside_ it.

Therefore, the upload message should not be present on top of every single block. In fact it should be present only on placeholders and galleries, where dropping a file on the block itself really _does_ insert the dropped file there.

GIF of behavior:

dragondrop

GIF of current behavior in `master`:

master

Note that this PR uses CSS to accomplish the above, which basically makes a whitelist of which blocks to show this message of. This doesn't seem ideal, and it also seems wasteful to include the actual text of the upload message even in paragraph blocks where they shouldn't be shown. @youknowriad any idea how to do this in a better way? 

Basically the desired behavior is that _for every block where the blue line shows up, we don't want the message_, whereas for every block where the blue line _doesn't_ show up, we want just the message.

### CODE REVIEW

**Comment by youknowriad** [comment]
I'm pretty sure, I had something in the Code that differentiate between the blocks that need this text or not. I think it's based on the presence of Drop Handlers. I'll have to take a look at how it worked before the original PR to remember how this was achieved :)

**Review by youknowriad — APPROVED** [review]
LGTM 👍

**Comment by jasmussen** [comment]
Lovely fix Riad, ship it!
